mint/git-clone/bin/git-clone

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env bash
+set -ueo pipefail
+
+mkdir -p "${CHECKOUT_PATH}"
+cd "${CHECKOUT_PATH}"
+
+if [[ "${PRESERVE_GIT_DIR}" == "true" ]]; then
+  CREDENTIAL_ARG=""
+  if [[ "${CREDENTIAL_HELPER}" != "" ]]; then
+    CREDENTIAL_ARG="-c credential.helper='${CREDENTIAL_HELPER}'"
+  fi
+  eval "git clone $CREDENTIAL_ARG \"${CHECKOUT_REPOSITORY}\" ."
+  git fetch origin "${CHECKOUT_REF}"
+  git checkout "${CHECKOUT_REF}"
+else
+  git init -b main
```

Comment:
`main` is arbitrary here -- this is just supplied to suppress a really annoying message about how git currently, but maybe will not always, uses `master` as the default branch by default